pkg/utils/eth.go

```diff
@@ -0,0 +1,51 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+func GWeiToWei(gwei float64) (*big.Int, error) {
+	if math.IsNaN(gwei) || math.IsInf(gwei, 0) {
+		return nil, fmt.Errorf("invalid gwei value: %v", gwei)
+	}
+
+	// convert float GWei value into integer Wei value
+	wei, _ := new(big.Float).Mul(
+		big.NewFloat(gwei),
+		big.NewFloat(params.GWei)).
+		Int(nil)
+
+	if wei.Cmp(abi.MaxUint256) == 1 {
+		return nil, errors.New("gwei value larger than max uint256")
+	}
+
+	return wei, nil
+}
+
+func EtherToWei(ether float64) (*big.Int, error) {
```

Comment:
Let's add some simple tests for those new methods?

Reply:
ok